packages/sdk/contractkit/src/kit.ts

```diff
@@ -187,14 +187,11 @@ export class ContractKit {
   getHumanReadableNetworkConfig = () => this.getNetworkConfig(true)
 
   /**
-   * Set CeloToken to use to pay for gas fees
-   * @param tokenContract CELO (GoldToken) or a supported StableToken contract
+   * Set an addressed to use to pay for gas fees
+   * @param address any string address
    */
-  async setFeeCurrency(tokenContract: CeloTokenContract): Promise<void> {
-    const address =
-      tokenContract === CeloContract.GoldToken
-        ? undefined
-        : await this.registry.addressFor(tokenContract)
+  setFeeCurrency(address: string) {
+    // TODO(Arthur): Should we define a HexString type to enforce it's an address?
```

Comment:
+1 to having  `type Address = '0x${string}'`  type

Reply:
There are two options: 

1. `` type Hex = `0x${string}` `` defined in `@celo/connect` > [types.ts](https://github.com/celo-org/developer-tooling/blob/f64bf9eef449a6e34048a50a16166bc0105d594f/packages/sdk/connect/src/types.ts#L11)
2. `` type StrongAddress = `0x${string}` `` defined in `@celo/base` > [address.ts](https://github.com/celo-org/developer-tooling/blob/f64bf9eef449a6e34048a50a16166bc0105d594f/packages/sdk/base/src/address.ts#L5)

Both are existing dependencies of contractkit, so they don't give or take on that front, but `@celo/base` has additional helper functions like [`isHexString()`](https://github.com/celo-org/developer-tooling/blob/f64bf9eef449a6e34048a50a16166bc0105d594f/packages/sdk/base/src/address.ts#L25) that could be quite handy to validate hex addresses.

For now, I'm choosing to use `@celo/base` here.

Reply:
nice!. 

i wonder since we have web3 available if using web3.utils.isAddress() would be good. benefits are 

A) less using our code more using standard

B) more specific not all hex strings are addresses

Reply:
Great find, let me take a closer look at `web3.utils.isAddress()`. Sounds like a good option 👍

Reply:
That's done in https://github.com/celo-org/developer-tooling/pull/151/commits/56f556785a2a9ecfd9f82d846548b7c2486ac34b